src/linear_algebra/affine_space/barycentric_coords.lean

```diff
@@ -0,0 +1,96 @@
+/-
+Copyright (c) 2021 Oliver Nash. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Oliver Nash
+-/
+import linear_algebra.affine_space.independent
+
+/-!
+# Barycentric coordinates
+
+Suppose `P` is an affine space modelled on the module `V` over the ring `k`, and `p : ι → P` is an
+affine-independent family of points spanning `P`. Given this data, each point `q : P` may be written
+uniquely as an affine combination: `q = w₀ p₀ + w₁ p₁ + ⋯` for some (finitely-supported) weights
+`wᵢ`. For each `i : ι`, we thus have an affine map `P →ᵃ[k] k`, namely `q ↦ wᵢ`. This family of
+maps is known as the family of barycentric coordinates. It is defined in this file.
+
+## The construction
+
+Fixing `i : ι`, and allowing `j : ι` to range over the values `j ≠ i`, we obtain a basis `bᵢ` of `V`
+defined by `bᵢ j = p j -ᵥ p i`. Let `fᵢ j : V →ₗ[k] k` be the corresponding dual basis and let
+`fᵢ = ∑ j, fᵢ j : V →ₗ[k] k` be the corresponding "sum of all coordinates" form. Then the `i`th
+barycentric coordinate of `q : P` is `1 - fᵢ (q -ᵥ p i)`.
+
+## Main definitions
+
+ * `barycentric_coord`
+ * `barycentric_coord_apply_eq`
+ * `barycentric_coord_apply_neq`
+ * `barycentric_coord_apply`
+ * `barycentric_coord_apply_combination`
+
+## TODO
+
+ * Construct the affine equivalence between `P` and `{ f : ι →₀ k | f.sum = 1 }`.
+
+-/
+
+open_locale affine big_operators
+open set
+
+universes u₁ u₂ u₃ u₄
+
+variables {ι : Type u₁} {k : Type u₂} {V : Type u₃} {P : Type u₄}
+variables [ring k] [add_comm_group V] [module k V] [affine_space V P]
+variables {p : ι → P} (h_ind : affine_independent k p) (h_tot : affine_span k (range p) = ⊤)
+include V h_ind h_tot
+
+/-- Given an affine-independent family of points spanning the point space `P`, if we single out one
+member of the family, we obtain a basis for the model space `V`. -/
```

Comment:
This could be more explicit, mentioning subtraction, especially since the definition is not easy to read.

Reply:
I had persuaded myself an opaque definition was OK in view of `basis_of_aff_ind_span_eq_top_apply` but I agree it's helpful (and free!) to add some more words here.